### PR TITLE
Add the integration tests to verify template version support in application management API

### PR DESCRIPTION
**Pull request by DilshanSenarath**
### Proposed changes in this pull request

> $subject

### Related Issues

- https://github.com/wso2/product-is/issues/20108

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/application/management/v1/ApplicationManagementSuccessTest.java [hunk 1]
```diff
@@ -111,6 +94,43 @@ public void createApplication() throws Exception {
         assertNotBlank(createdAppId);
     }
 
+    @Test(dependsOnMethods = {"createApplication"})
```
Comment (ImalshaG) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/application/management/v1/ApplicationManagementSuccessTest.java:97]: Let's leave the existing testGetAllApplications() method as it is to test the default application list api and introduce a new method to test the get application list api with required params (including the template id and version) in an extendable manner.

**Review by ImalshaG — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -2272,7 +2272,7 @@
     <properties>
 
         <!--Carbon Identity Framework Version-->
-        <carbon.identity.framework.version>7.3.41</carbon.identity.framework.version>
+        <carbon.identity.framework.version>7.3.41-SNAPSHOT</carbon.identity.framework.version>
```
Comment (pavinduLakshan) [inline pom.xml:2275]:
```suggestion
        <carbon.identity.framework.version>7.3.41</carbon.identity.framework.version>
```

File: pom.xml [hunk 3]
```diff
@@ -2385,7 +2385,7 @@
         <!-- Identity REST API feature -->
         <identity.api.dispatcher.version>2.0.17</identity.api.dispatcher.version>
         <identity.user.api.version>1.3.38</identity.user.api.version>
-        <identity.server.api.version>1.2.210</identity.server.api.version>
+        <identity.server.api.version>1.2.211-SNAPSHOT</identity.server.api.version>
```
Comment (pavinduLakshan) [inline pom.xml:2388]:
```suggestion
        <identity.server.api.version>1.2.211</identity.server.api.version>
```

Comment (DilshanSenarath) [reply pom.xml:2388]: This will be updated when the relevant component is released.

File: pom.xml (hunk 2 above)
Comment (DilshanSenarath) [reply pom.xml:2275]: This will be updated when the relevant component is released.

**Review by AnjanaSamindraPerera — APPROVED** [review]

**Pull request merged**